inductiva/tests/tasks/test_run_task.py

```diff
@@ -0,0 +1,51 @@
+"""Test file for the run_simulation file."""
+import os
+import json
+
+import pathlib
+
+from inductiva import tasks
+
+LOGGED_TASK_ID = "1234"
+NOT_LOGGED_TASK_ID = "4321"
+
+
+def _id_in_metadata_file(task_id):
+    """Checks if the ID exists in the file."""
+    with open(tasks.TASK_METADATA_FILENAME, "r", encoding="utf-8") as f:
+        json_list = [json.loads(line) for line in f]
+    return task_id in map(lambda x: x["task_id"], json_list)
+
+
+def test_run_simulation_logging(mocker):
+    """Tests if the id of the task was added to the file."""
+
+    # Old logs may cause problems so we remove the file.
+    if os.path.exists(tasks.TASK_METADATA_FILENAME):
+        os.remove(tasks.TASK_METADATA_FILENAME)
+
+    invoke_api_mock = mocker.patch("inductiva.api.methods.invoke_async_api")
+    mocker.patch("inductiva.tasks.Task")
```

Comment:
Why is this patch necessary here?

Reply:
We mock the `Task` to prevent this line of code from happening:

```
task = tasks.Task(task_id)
```

Which would make several call for the API. I believe that we would even have to validate an API which should not happen during this test.

Reply:
yea, just patch what is strictly required.
sometimes, rather than patching we can use "inversion of control". In this case, it means changing the `tasks.run_simulation` method to receive, as an argument, the `methods.invoke_async_api` method:

```python
def tasks.run_simulation(..., api_invoker=None):
    if api_invoker is None:
        api_invoker = methods.invoke_async_api
   ...
    task_id = api_invoker(...)
```

In the test, you'd then call the `tasks.run_simulation` passing it you api mock. No need to patch and nothing else can break due to patching.